### PR TITLE
DEVDOCS-2754: Add Webhook Port Limitations

### PR DESCRIPTION
**Pull request by aglensmith**
# [DEVDOCS-2754](https://jira.bigcommerce.com/browse/DEVDOCS-2754)

## What changed?
* Added notes about custom ports not being supported. 
* Re-wrote and updated webhook tutorial to use the ngrok npm package for simplicity. 

I tested the new instruction for the webhooks tutorial on my windows machine using git bash. They worked nicely.

### CODE REVIEW

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 1]
```diff
@@ -4,431 +4,159 @@
...
 ### On this page
 
-- [Prerequisites](#prerequisites)
-- [Create project folder](#create-project-folder)
-- [Install ngrok](#install-ngrok)
-- [Create express app](#create-express-app)
-- [Start the app and ngrok](#start-the-app-and-ngrok)
+- [Create an Express app](#create-an-express-app)
+- [Start the app](#start-the-app)
+- [Create a webook](#create-a-webook)
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:9]:
```suggestion
- [Create a webhook](#create-a-webhook)

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 2]
```diff
@@ -4,431 +4,159 @@
...
 - [Trigger the webhook event](#trigger-the-webhook-event)
 - [Adding custom headers](#adding-custom-headers)
 - [Troubleshooting](#troubleshooting)
 - [Resources](#resources)
 
 </div>
 
-In this tutorial, we'll install the software ngrok, register a BigCommerce webhook on your store, and then observe the response when the webhook event is triggered.
+In this tutorial, we'll create a Node.js Express app that handless webhook callbacks and uses [ngrok](https://ngrok.com/) (ngrok.com) to expose the app to the internet. Then, we'll create a webhook and observe the callback using the ngrok web interface when the event is triggered.
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:17]:
```suggestion
In this tutorial, we'll create a Node.js Express app that handless webhook callbacks and uses [ngrok](https://ngrok.com/) (ngrok.com) to expose the app to the Internet. Then, we'll create a webhook and observe the callback using the ngrok web interface when the event is triggered.

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 3]
```diff
@@ -4,431 +4,159 @@
...
 
-[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/3f005ed74030e01bbf7a)
+- [API Access Token](https://developer.bigcommerce.com/api-docs/getting-started/authentication/rest-api-authentication) with [scope](https://developer.bigcommerce.com/api-docs/getting-started/authentication/rest-api-authentication#oauth-scopes) set to **Information & Settings** read-only and **Products** read-only.
+- [Webhooks Overview](https://developer.bigcommerce.com/api-docs/store-management/webhooks/overview)
+- Familiarity with working in the terminal
+- Familiarity working with `node` and `npm`
+## Create an Express app
 
-## Prerequisites
+First, let's make a `webooks-test` directory and initialize a Node.js Express app inside of it. To do so, run the following commands in a terminal.
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:27]:
```suggestion
First, let's make a `webooks-test` directory and initialize a Node.js Express app inside of it. To do so, run the following commands in the terminal.

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 4]
```diff
@@ -4,431 +4,159 @@
...
-- [API Access Token](/api-docs/getting-started/basics/authentication#authentication_getting-api-credentials) with [scope](/api-docs/getting-started/basics/authentication#authentication_oauth-scopes) set to Information & Settings read-only and Products read-only.
-- [Webhooks Overview](/api-docs/getting-started/webhooks/about-webhooks)
+```bash
+mkdir webhooks-test         # Create project directory
 
-## Create project folder
+cd webhooks-test            # Move into project directory
 
-1. Open the terminal and create a folder, then move into that directory. This folder will hold ngrok and the Node app. 
+npm init                    # Initialize node app (hit return to accept the default values)
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:34]:
```suggestion
npm init                    # Initialize a Node.js project (hit return to accept the default values)

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 5]
```diff
@@ -4,431 +4,159 @@
...
-<!--
-title: ""
-subtitle: ""
-lineNumbers: true
--->
-
-**Downloading ngrok and moving contents to folder using the shell**
-```shell
-unzip /Users/your-computer/Downloads/ngrok-stable-darwin-amd64.zip -d /Users/your-computer/Documents/webhooks-test
+touch index.js              # create main app script file
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:40]:
```suggestion
touch index.js              # Create main app script file

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 6]
```diff
@@ -4,431 +4,159 @@
...
-
-### Checkpoint
-> At this point you should have a project folder with ngrok unzipped inside of it. 
-
-</div>
-</div>
-</div>
-
-## Create express app
+Then, copy and paste the JavaScript below into `index.js`.
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:43]:
```suggestion
Then, copy and paste the following JavaScript code into `index.js`.

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 7]
```diff
@@ -4,431 +4,159 @@
...
-Listening for webhooks on port 3000
-```
+<div class="HubBlock--callout">
+<div class="CalloutBlock--info">
+<div class="HubBlock-content">
 
-#### ngrok http 3000
+> ### Note
+> * Ngrok is a helpful tool for viewing webhook callbacks BigCommerce sends to your app. Ngrok creates a publicly accessible tunnel URL to an application running on your machine. When using ngrok you can view HTTP request details in its web interface.
+> * For simplicity, this tutorial uses an [npm package](https://www.npmjs.com/package/ngrok) to install and run ngrok. For official ngrok usage and installation instructions, [visit ngrok.com](https://ngrok.com/).
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:77]:
```suggestion
> * For simplicity, this tutorial uses an [npm package](https://www.npmjs.com/package/ngrok) to install and run ngrok. For official ngrok usage and installation instructions, visit [ngrok.com](https://ngrok.com/).

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 8]
```diff
@@ -4,431 +4,159 @@
...
+</div>
+</div>
 
-ngrok returns two values we will need to register a webhook and observe the response: 
-- **Web Interface:** Where you can monitor the hook from a browser. `http://127.0.0.1:4040`
-- **Forwarding:** The webhook destination. Make note of the https version. For example, `https://6a35e97b.ngrok.io`.
+## Start the app
 
-4. Open a web browser and copy in your Web Interface URL: `http://127.0.0.1:4040`.  This is where webhook payloads will appear when they fire.
+To start the app, run the following commands.
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:85]:
```suggestion
To start the app, run the following commands:

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 9]
```diff
@@ -4,431 +4,159 @@
...
-title: ""
-subtitle: ""
-lineNumbers: true
--->
-**Webhook POST request body**
-```json
 {
  "scope": "store/product/updated",
- "destination": "https://6a35e97b.ngrok.io/webhooks",
+ "destination": "https://6a35e97b.ngrok.io/webhooks", # replace 6a35e97b.ngrok.io with your HTTPS tunnel URL
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:107]:
```suggestion
 "destination": "https://6a35e97b.ngrok.io/webhooks", # Replace 6a35e97b.ngrok.io with your HTTPS tunnel URL

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 10]
```diff
@@ -4,431 +4,159 @@
...
-  "price": 12.99
-}
-```
-
-After hitting send, check the ngrok web interface. You may see a single event or several based on how many times you updated the product in the previous step.
+## Trigger the webhook event
 
-<!--
-    title: #### ngrok Web Interface
+Webhooks fire when shoppers perform actions on the storefront and when users make changes in the control panel. They also fire when updates are made via API. Trigger the webhook you just created by performing the following actions in your BigCommerce control panel.
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:128]:
```suggestion
Webhooks fire when shoppers perform actions on the storefront and when users make changes in the control panel. They will also fire when you make changes using an API. Trigger the webhook you just created by performing the following actions in your BigCommerce control panel:

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 11]
```diff
@@ -437,44 +165,34 @@ lineNumbers: true
...
-    </div><div class="HubBlock-header-subtitle"></div>
-</div>
+**Getting a `404` error using the root (/) url?**
 
-<!--
-title: ""
-subtitle: ""
-lineNumbers: true
--->
+Add this snippet to your code to respond to incoming get requests with 'hello':
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:179]:
```suggestion
Add this snippet to your code to respond to incoming `GET` requests with 'hello':

```

File: docs/api-docs/getting-started/webhooks/setting-up-webhooks.md [hunk 12]
```diff
@@ -437,44 +165,34 @@ lineNumbers: true
...
-Use the command `mv ngrok /usr/local/bin `to move ngrok to your local bin folder. This way it becomes available globally. 
-Use the command `./ngrok http 3000` to run ngrok as a sudo user. 
-
+Use the command `mv ngrok /usr/local/bin `to move ngrok to your local bin folder. This way it becomes available globally.
+Use the command `./ngrok http 3000` to run ngrok as a sudo user.
 
 **Windows users**
 
-If you are having trouble getting ngrok started try setting the PATH. 
+If you are having trouble getting ngrok started try setting the PATH.
```
Comment (ektravel) [inline docs/api-docs/getting-started/webhooks/setting-up-webhooks.md:195]:
```suggestion
If you are having trouble getting ngrok started, try setting the PATH.

```

**Review by ektravel — COMMENTED** [review]
Added my suggestions.

**Review by uwikaiddi — APPROVED** [review]

**Pull request merged**